diff --git a/Makefile b/Makefile
index 2995f491b..ed9b30a2c 100644
--- a/Makefile
+++ b/Makefile
@@ -131,7 +131,7 @@ endif
 # Figure out optimize level.
 ifneq ($(DEBUG_LEVEL), 2)
 ifeq ($(LITE), 0)
-	OPTIMIZE_LEVEL ?= -O2
+	OPTIMIZE_LEVEL ?= -O3
 else
 	OPTIMIZE_LEVEL ?= -Os
 endif
@@ -142,7 +142,7 @@ OPT += $(OPTIMIZE_LEVEL)
 
 # compile with -O2 if debug level is not 2
 ifneq ($(DEBUG_LEVEL), 2)
-OPT += -fno-omit-frame-pointer
+OPT += -fomit-frame-pointer
 # Skip for archs that don't support -momit-leaf-frame-pointer
 ifeq (,$(shell $(CXX) -fsyntax-only -momit-leaf-frame-pointer -xc /dev/null 2>&1))
 OPT += -momit-leaf-frame-pointer
